### ISSUE TITLE
How to locate which value is wrong?

### ISSUE THREAD

**Issue by sunxd-lab** [capnproto/capnproto#2186]
I'm using v1.0.2, this is my code:
`
# example.capnp

@0xbf5147cf9c9a31ab;

struct Example {
  id @0 :Int32;
  name @1 :Text;
  values @2 :Int32;
}

# example.json

{
    "id": "12",
    "name": "capnp",
    "values": false
}
`
when I compile:
`
capnp convert json:packed example.capnp Example < example.json --verbose
`

show it:
`
*** Uncaught exception ***
capnp/compat/json.c++:446: failed: Expected integer value
stack: 55a5fd485f77 55a5fd485b26 55a5fd492f60 55a5fd491d5c 55a5fd48e168 55a5fd485c72 55a5fd48395d 55a5fd35c237 55a5fd35a8bc 55a5fd38e07d 55a5fd38e0d8 55a5fd38e12d 55a5fd585694 55a5fd5805f6 55a5fd58ca58 55a5fd58388c 55a5fd57fa6c 55a5fd58ca58 55a5fd58388c 55a5fd57d3ac
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
    ??:0: returning here
`

I known "values" should be integer, but when I have a large amount of data, how do I locate it?

**Comment by kentonv** [comment]
Yeah the error handling in JsonCodec could use a lot of work. In fact, the whole class could use a lot of work, e.g. optimization. But I don't really have bandwidth to work on this, sorry.
